core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedClusterMessage.scala

```diff
@@ -36,9 +36,13 @@ private[spark] object CoarseGrainedClusterMessages {
   case class KillTask(taskId: Long, executor: String, interruptThread: Boolean)
     extends CoarseGrainedClusterMessage
 
-  case object RegisteredExecutor extends CoarseGrainedClusterMessage
+  sealed trait RegisterExecutorResponse
+
+  case class RegisteredExecutor() extends CoarseGrainedClusterMessage
```

Comment:
because it needs to extend the trait, and it didn't work when I did not make that change. (I also have a change built on top of this one where I'll have to add a parameter to this message.)